### ISSUE TITLE
Draft: some ideas. I will update it frequently. And will remove draft when i finish.

### ISSUE THREAD

**Issue by matart15** [nstaldev/nstal#2]
- [ ] abiliity to change part of the commands
    lets say we have this command `npx create-next-app my-nstal-and-mdx-blog`. User may want to change `npx create-next-app some-other-name`
- [ ] component for Dangerous commands. or flag

**Comment by matart15** [comment]
@phbernard let me know what do you think.

**Comment by phbernard** [comment]
Thank you for all this feedback! I already had some of these points in mind, which is great since it means I'm not the only one thinking that way. Some other are new to me, which is great too, they make the project progress!

1. Interesting point. I submitted #6
2. I would rephrase this one with: "each action should have a good and comprehensive UX". I entered #7, but I guess each action UX should be carefully crafted.
3. That's right. #8 and #9.
4. I'm not sure about this one. nstal is designed to be SEO friendly. In other words, an nstaller should be as Google-friendly as any regular blog post. If some content is collapsed by default, it will be invisible to Google (or Google will know it's not shown, which leads to the same outcome) and the hidden text won't count from an SEO standpoint. I'd rather prefer to address this later, when we have more feedback about how nstallers are used and how they should look like.
5. Yep! Would you submit an issue for this?
6. Oh yeah! That's #10.
7. You're right! Could you submit an issue for this?
8. Not as trivial as it sounds. nstal CLI opens a connection available for everyone. It takes a password so only the nstaller can connect. This is why the command to start `ntal` is something like `npx nstal connect XtiUnCW6B2` and not just `npx nstal connect`. So the question is: do the nstaller still have this code, for example on a page refresh? (currently, no, a page refresh just erase the React state). But maybe I misunderstood you point. If that's so, please submit a separate issue and elaborate.
9. Definitely! Would you submit an issue?
10. Clearly! First, the nstaller should receive command outputs so it can process them: display them, check a particular message, etc. The nstaller should also receive the command status so it can provide feedback to the user. I consider this point to be part of #8. For example, if the `cd abc` command fails because `abc` is not a directory but a file, the action should says just that.

I propose we close this issue as soon as we have created as many issues as needed to discuss these points in separate threads.

**Comment by matart15** [comment]
1. ✅ 
2. ✅ 
3. ✅ 
4. ✅ `nstal is designed to be SEO friendly.`. Actially I did not think about that point. 🆗
5. ✅ https://github.com/nstaldev/nstal/issues/12
6. ✅ 
7. ✅ https://github.com/nstaldev/nstal/issues/13
8. Will write detail in separate comment.
9. ✅ https://github.com/nstaldev/nstal/issues/14
10. Will write detail in separate comment.

**Comment by matart15** [comment]
10. My concern was if `reader` is reading documentation there is high chance that `reader` have very limit understanding of what should happen after this command. So `Reader` may continue on wrong state and fall into bad situation.

How about `FOR NOW` we just implement Simple component that asks user and go forward only if user clicks `yes`.

example

Flow: 
  1. commands: [ `cd abc`, `pwd` ]
  2. `Do you see abc at the end?`. `yes` `no`
    a. if `no` kindly remind reader that `next commands require you to be in "abc" dir`. or `try "ls -al"`
    b. if `yes` we move forward 

Component
```ts
<RunCommandsWithConfirmation
  commands={[
    "cd abc"
    "pwd"
  ]}
  confirmationMessage="Do you see abc at the end?"
  failExplantion="try `ls -al`"
/>
